packages/cheqd/src/anoncreds/services/CheqdAnonCredsRegistry.ts

```diff
@@ -305,24 +366,35 @@ export class CheqdAnonCredsRegistry implements AnonCredsRegistry {
     try {
       const cheqdDidResolver = agentContext.dependencyManager.resolve(CheqdDidResolver)
       const parsedDid = parseCheqdDid(revocationRegistryId)
-      if (!parsedDid) {
-        throw new Error(`Invalid revocationRegistryId: ${revocationRegistryId}`)
-      }
+      if (!parsedDid) throw new CredoError(`Invalid revocationRegistryId: ${revocationRegistryId}`)
 
       agentContext.config.logger.trace(
         `Submitting get revocation status request for '${revocationRegistryId}' to ledger`
       )
 
+      const revocationRegistryDefinition = await this.getRevocationRegistryDefinition(
+        agentContext,
+        revocationRegistryId
+      )
+      if (!revocationRegistryDefinition.revocationRegistryDefinition)
+        throw new CredoError(`Revocation registry definition not found for id: ${revocationRegistryId}`)
+
+      const revocationRegistryDefinitionName = revocationRegistryDefinition.revocationRegistryDefinitionMetadata.name
+      if (!revocationRegistryDefinitionName)
+        throw new CredoError(`Revocation registry definition name not found for id: ${revocationRegistryId}`)
+
       const response = await cheqdDidResolver.resolveResource(
         agentContext,
-        `${revocationRegistryId}&resourceType=anonCredsStatusList&resourceVersionTime=${timestamp}`
+        `${parsedDid.did}?resourceType=anonCredsStatusList&resourceVersionTime=${timestamp}&resourceName=${revocationRegistryDefinitionName}`
```

Comment:
Does the revocation registry definition name need to be unique? couldn't we use the revocationRegistryId as the name of the status list?

Reply:
Based on the documentation of cheqd about anoncreds revocation. The statuslist and the revocation registry definition name needs to be the same as the credential definition name. Which is currently the case in the implementation.